src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java

```diff
@@ -154,15 +178,16 @@ private void visitObject(MaskingState maskingState, @Nullable KeyMaskingConfig p
                 break;
             }
             // In case target keys should be considered as allow list, we need to NOT mask certain keys
-            int openingQuoteIndex = maskingState.currentIndex();
+            maskingState.registerValueStartIndex();
 
             stepOverStringValue(maskingState);
 
             int afterClosingQuoteIndex = maskingState.currentIndex();
-            int keyLength = afterClosingQuoteIndex - openingQuoteIndex - 2; // minus the opening and closing quotes
-            maskingState.expandCurrentJsonPath(keyMatcher.traverseJsonPathSegment(maskingState.getMessage(), maskingState.getCurrentJsonPathNode(), openingQuoteIndex + 1, keyLength));
-            KeyMaskingConfig keyMaskingConfig = keyMatcher.getMaskConfigIfMatched(maskingState.getMessage(), openingQuoteIndex + 1, // plus one for the opening quote
+            int keyLength = afterClosingQuoteIndex - maskingState.getCurrentValueStartIndex() - 2; // minus the opening and closing quotes
```

Comment:
This change is very odd. Keylength was the after close index minus the open and now it is after close index minus the `maskingState.getCurrentValueStartIndex`. I think this makes the whole code much less readible to understand that this is because this is the index registered in ` maskingState.registerValueStartIndex()`. It is not the start of a current value, it is the start of the key in this case.

Reply:
The idea of the change is not to expose currentIndex anywhere outside of MaskingState so that we are sure that we can safely reset the indexes because there is no one "outside" who remembers old indexes while the next buffer is being read.
It is a good point that this is a start of the key and not the start of the current value. I will rename the method to `registerTokenStartIndex` as this new name reflects both registering value start and key start indexes.

Reply:
Done